sdk/core/azure-core/src/http/logging_policy.cpp

```diff
@@ -30,22 +30,69 @@ std::string TruncateIfLengthy(std::string const& s)
   return s.substr(0, BeginLength) + Ellipsis + s.substr(length - EndLength, EndLength);
 }
 
-std::string GetRequestLogMessage(Request const& request)
+std::string GetRequestLogMessage(
+    Azure::Core::Http::LoggingPolicyOptions const& options,
+    Request const& request)
 {
   std::ostringstream log;
 
-  log << "HTTP Request : " << HttpMethodToString(request.GetMethod()) << " "
-      << request.GetUrl().GetAbsoluteUrl();
+  std::string url = request.GetUrl().GetAbsoluteUrl();
+  if (!url.empty())
+  {
+    auto qm = url.find('?');
+    if (qm != std::string::npos)
+    {
+      url = url.substr(0, qm);
```

Comment:
Feel free to update the Url class and add a `HasQueryParameters()` method which can save you from calling `find()` here.

I would also recommend to add a `GetShortUrl()` or something which would return only the url up to the query parameters start (if any). Then you can just call GetQueryParameters.

We can use the Url for our own benefit here